### DISCUSSION TITLE
[Feature] Allow "angle" for EnvShake sctrl.

### DISCUSSION THREAD

**Discussion by Hexioum** [ikemen-engine/Ikemen-GO#941] [Feature Request] [answered]
As it is now, every screenshake is only vertical, but there is other fighting games (BBCF, P4AU) that allow the screenshakes to have different angles.

**Describe the solution you'd like**
Add a parameter for the EnvShake sctrl that is "angle".
**angle = 0** is the default parameter, means the shake is vertical.
**angle = 45** diagonal-up envshake. Diag-up and then diag-down in the opposite direction.
**angle = 90** horizontal screenshake. Always to the right side of the screen.
**angle = 180** would be the same as **angle = 0** if you chose a negative ampl.

Another example,
```
type = envShake
trigger1 = moveHit = 1
time = 12
ampl = 6
angle = 90*facing
ignoreHitPause = 1
persistent = 0
```
Means that the screen will always move first in the direction of the player that made the attack.

**Additional context**
This is another game that has this feature. It really gives a lot more impact to the interactions.
https://www.youtube.com/watch?v=tspf-Z6tbGY&t=22

**Comment by potsmugen** [comment]
This would be cool. The same enhancement could be copy pasted in the Hitdef Envshake. They should probably go counter-clockwise like everything else (I think) though.

**Comment by potsmugen** [comment, marked as answer]
Already implemented with `dir` parameter.
